### ISSUE TITLE
reset::hard::resetting_reverts_unmerged depends on state from previous tests

### ISSUE THREAD

**Issue by joshtriplett** [libgit2/libgit2#3726]
Running `./libgit2_clar -v -sreset::hard::resetting_reverts_unmerged` , or the full reset suite with `./libgit2_clar -v -sreset`, produces the following failure:

```
  1) Failure:
reset::hard::resetting_reverts_unmerged [/home/josh/src/libgit2/tests/reset/hard.c:115]
  Function call failed: (git_index_add(index, &entry))
  error -1 - Object not found - cannot read header for (6bb0d9f700543ba3d318ba7075fc3bd696b4287)
```

This failure doesn't occur when running the full test suite.  I think something in this test depends on state set up by a previous test.

**Comment by carlosmn** [comment]
It looks like we're accidentally leaving strict object creation off, so this test was left out of the fix when activating it by default.

**Comment by carlosmn** [comment]
Fixed via #3737
